libenv/sysinfo.c

```diff
@@ -3465,6 +3465,93 @@ static void SysOSNameHuman(EvalContext *ctx)
 
 /*****************************************************************************/
 
+/**
+ * Extract version number from string in place.
+ * 
+ * @param [in]  str string to extract version number from
+ * @param [out] num pointer to start of version number or %NULL if no version
+ *                  number was found
+ * 
+ * @return pointer to the remaining string in `str` or %NULL if no remainder
+ * 
+ * @note `str` will be mutated
+ */
+static char *ExtractVersionNumber(char *str, char **num)
```

Comment:
This function does not have much to do with version numbers. (It doesn't even care about dots, dashes, or spaces).

Maybe call it `FindNextInteger()`?